BluetoothLeChat/app/src/main/java/com/example/bluetoothlechat/scan/DeviceScanViewModel.kt

```diff
@@ -81,7 +84,7 @@ class DeviceScanViewModel(app: Application) : AndroidViewModel(app) {
             _viewState.value = ActiveScan
 
             // Stop scanning after the scan period
-            Handler().postDelayed({ stopScanning() }, SCAN_PERIOD)
+            Handler(Looper.myLooper()!!).postDelayed({ stopScanning() }, SCAN_PERIOD)
```

Comment:
Convert to coroutines. We aren't supposed to be using the other stuff.

Reply:
done